### PR TITLE
Add v6 version of ml-slurm blueprint and integration test

### PR DESCRIPTION
**Pull request by harshthakkar01**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/ml-slurm-v6.yaml [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    use:
+    - network
+    - script
+    settings:
+      # give VM a public IP to ensure startup script can reach public internet
+      # w/o new VPC
+      omit_external_ip: false
+      source_image_project_id: [schedmd-slurm-public]
+      # see latest in https://github.com/GoogleCloudPlatform/slurm-gcp/blob/master/docs/images.md#published-image-family
+      source_image_family: slurm-gcp-6-4-hpc-rocky-linux-8
```
Comment (nick-stroud) [inline examples/ml-slurm-v6.yaml:150]: The v5 version uses Debian as the base image. I think that is kind of important to the example. Why has it changed?

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: examples/ml-slurm-v6.yaml (hunk 1 above)
Comment (nick-stroud) [reply examples/ml-slurm-v6.yaml:150]: PS. If we use Debian then this example would be one to add to the support matrix for slurm + startup script + debian

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: examples/ml-slurm-v6.yaml (hunk 1 above)
Comment (harshthakkar01) [reply examples/ml-slurm-v6.yaml:150]: No specific reason other than its default for SlurmGCP v6. I will update it to use debian.

Comment (harshthakkar01) [reply examples/ml-slurm-v6.yaml]: Updated to debian. For matrix I created new task and will update for everything at once.

File: examples/ml-slurm-v6.yaml
Comment (nick-stroud) [inline examples/ml-slurm-v6.yaml:1]: Please have @tpdownes take a look at the blueprint.

**Review by nick-stroud — APPROVED** [review]

File: examples/ml-slurm-v6.yaml [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+      - type: shell
+        destination: install-ml-libraries.sh
+        content: |
+          #!/bin/bash
+          # this script is designed to execute on Slurm images published by SchedMD that:
+          # - are based on Debian 11 distribution of Linux
+          # - have NVIDIA Drivers v530 pre-installed
+          # - have CUDA Toolkit 12.1 pre-installed.
+
+          #set -e -o pipefail
```
Comment (tpdownes) [inline examples/ml-slurm-v6.yaml:80]: I think you unintentionally commented `set -e -o pipefail` out.

File: examples/ml-slurm-v6.yaml [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    use:
+    - network
+    - script
+    settings:
+      # give VM a public IP to ensure startup script can reach public internet
+      # w/o new VPC
+      omit_external_ip: false
+      source_image_project_id: [schedmd-slurm-public]
+      # see latest in https://github.com/GoogleCloudPlatform/slurm-gcp/blob/master/docs/images.md#published-image-family
+      source_image_family: slurm-gcp-6-4-debian-11
```
Comment (tpdownes) [inline examples/ml-slurm-v6.yaml:150]: I can't see how this script is going to succeed if you use the Debian 11 images. The startup script appears to be tuned for Rocky 8

**Review by tpdownes — CHANGES_REQUESTED** [review]

File: examples/ml-slurm-v6.yaml
Comment (tpdownes) [reply examples/ml-slurm-v6.yaml:1]: Done below

File: examples/ml-slurm-v6.yaml (hunk 2 above)
Comment (harshthakkar01) [reply examples/ml-slurm-v6.yaml]: Thanks for pointing this out. Updated.

File: examples/ml-slurm-v6.yaml (hunk 3 above)
Comment (harshthakkar01) [reply examples/ml-slurm-v6.yaml:142]: Thanks for noticing important point. Updated.

**Review by tpdownes — APPROVED** [review]
Please merge after the rebase build passes. LGTM.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**